lib/attrsets.nix

```diff
@@ -419,18 +419,44 @@ rec {
     set:
     listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [(nameValuePair name v)] else []) (attrNames set));
 
+  /*
+    An alias for [`filterAttrsRecursiveTopDown`](#function-library-lib.attrsets.filterAttrsRecursiveTopDown).
+  */
+  filterAttrsRecursive = filterAttrsRecursiveTopDown;
 
-  /* Filter an attribute set recursively by removing all attributes for
-     which the given predicate return false.
+  /*
+    Filter an attribute set recursively by removing all attributes for which the given predicate returns `false`.
 
-     Example:
-       filterAttrsRecursive (n: v: v != null) { foo = { bar = null; }; }
-       => { foo = {}; }
+    More precisely, for each attribute:
+    - Only include the attribute if the predicate returns `true` for this attribute.
+    - If the attribute is included and its value is an attribute set itself,
+      replace it with the result of calling `filterAttrsRecursiveTopDown` on it.
 
-     Type:
-       filterAttrsRecursive :: (String -> Any -> Bool) -> AttrSet -> AttrSet
+    Compared to [`filterAttrsRecursiveBottomUp`](#function-library-lib.attrsets.filterAttrsRecursiveBottomUp):
+    - This function allows the predicate to stop the recursion based on the value.
```

Comment:
Doesn't that depend on the strictness of the predicate? A predicate that only looks at the name is useful and not strict in `v`, so I don't see why it would be evaluated.